Java: add dataflow-generated models for JDK17

owen-mc opened the pull request:
Add dataflow-generated models for JDK17 and update test expectations as needed.

Note that preparatory work was done in:
- [this PR](https://github.com/github/codeql/pull/15280) (adding manual models to override dataflow-generated models that didn't combine well with some existing manual models, and also adding neutral summary models for some callables that we don't want summary models for)
- [this PR](https://github.com/github/codeql/pull/15246) (making manual neutral summary models block dataflow-generated summary models)
- [this PR](https://github.com/github/codeql/pull/15179) (don't generate models that won't be used because the are blocked by a manual model, and also don't generate summary and neutral models for the same callable).
- [this PR]( https://github.com/github/codeql/pull/15766) (add some neutral models)
- [this PR]( https://github.com/github/codeql/pull/15767) (add some summary models and some neutral models)
- [this PR](https://github.com/github/codeql/pull/15906) (add some neutral models)
- [Java: more manual models](https://github.com/github/codeql/pull/15946)
- [ Java: Add more neutrals and improve java.net.URL models](https://github.com/github/codeql/pull/16062)

Note to reviewers: obviously you can't look through all the generated models. If you want you can do some spot checking to make sure things look okay. But I think it's okay to just look at the other commits and makes sure that the test changes do look reasonable given the new models. If we find that some of the generated models are wrong then we can add manual models for those callables, but that shouldn't stop us from merging these generated models that we believe are significantly correct. If there are improvements to the model generator in future then we can regenerate models for the JDK and benefit from improved models.

coadaflorin commented on java/ql/lib/change-notes/2024-03-19-many-generated-jdk-models.md:
```diff
@@ -0,0 +1,4 @@
+---
+category: minorAnalysis
+---
+* About 15,000 summary models and neutral summary models for the JDK that were generated using data flow have been added. This may lead to new results being reported.
```
We generally use alerts instead of results in most communications.

owen-mc commented:
Rebasing on https://github.com/github/codeql/pull/16062 made a test fail. I have fixed that and also updated the commits which delete generated models which have been deliberately/incidentally modeled since I generated models. This is ready to review and hopefully merge soon.

smowton reviewed (approved):
Looked at evaluation results, which appear positive

coadaflorin commented:
Is this going to be in 2.17.1? If so, I'll prep some wording for that release changelog.

owen-mc commented:
I believe so.
